src/node.rs

```diff
@@ -369,35 +354,40 @@ impl<P: KeyTrait + Clone, N: Version, const WIDTH: usize> Drop for FlatNode<P, N
 pub struct Node48<P: KeyTrait + Clone, N: Version> {
     pub(crate) prefix: P,
     pub(crate) version: u64,
-    keys: BitArray<u8, 256>,
-    children: BitArray<Arc<N>, 48>,
-    num_children: u8,
+    keys: Box<[u8; 256]>,
+    children: Box<[Option<Arc<N>>; 48]>,
+    child_bitmap: u64,
```

Comment:
This is a bit map keeping track of the zeros. Ones indicate that child is filled and zeroes indicate empty spots.